azure/Kqlmagic/my_aad_helper.py

```diff
@@ -92,9 +93,14 @@ def __init__(self, kcsb, default_clientid, **options):
         client_id = kcsb.application_client_id or default_clientid
         self._resource = "{0.scheme}://{0.hostname}".format(urlparse(kcsb.data_source))
         token_cache = None
-        isSso = "FALSE" # os.getenv("{0}_ENABLE_SSO".format(Constants.MAGIC_CLASS_NAME.upper()))
-        if (isSso and isSso.upper() == "TRUE"):
-            token_cache = AdalTokenCache()
+
+        SSO_encrypt_keys = AdalTokenCache.get_params_SSO()
+        if SSO_encrypt_keys[0] and options.get("enablesso"):
+            username_SSO = SSO_encrypt_keys[1]
+            secret_key_SSO = SSO_encrypt_keys[2]
+            salt_bytes = SSO_encrypt_keys[3]
+            token_cache = AdalTokenCache(username_SSO, secret_key_SSO, salt_bytes, options.get("tokencleanuptime"))
```

Comment:
I suggest that AdalTokenCache.get_params_SSO() will do the code of line 98 till line 101 and will return either None or a dictionary with key/values (including options.get("tokencleanuptime")). if it return None then we don't do caching, if it returns a dictionary, just pass the dictionary as a parameter to AdalTokenCache.